### PR TITLE
Feature/poetry tasks

### PR DESCRIPTION
**Pull request by JulesBelveze**
This PR aims at adding poetry tasks and integrating them into the CI.

## Checklist:

- [ ] I've added Google style docstrings to my code.
- [ ] I've used `pydantic` for typing when/where necessary.
- [ ] I have linted my code
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: langtest/utils/custom_types/sample.py
Comment (chakravarthik27) [inline langtest/utils/custom_types/sample.py:1]: Hi @JulesBelveze,

can you please look into this variable? this variable is same in utils.py.
In past, I faced **Circular Import** Error. I don't know how to resolve it.

Comment (JulesBelveze) [reply langtest/utils/custom_types/sample.py:1]: I'll have a look at it. Is there any reason why we have the same variable name twice?

Comment (chakravarthik27) [reply langtest/utils/custom_types/sample.py:1]: Circular Import Error

Comment (JulesBelveze) [reply langtest/utils/custom_types/sample.py:1]: No but I mean why do we need that variable defined in two separate places

Comment (chakravarthik27) [reply langtest/utils/custom_types/sample.py:1]: to avoid circular Import error

**Pull request merged**